src/wallet/wallet.cpp

```diff
@@ -4939,9 +4943,9 @@ CAmount CWalletTx::GetCredit(const isminefilter& filter) const
     return credit;
 }
 
-CAmount CWalletTx::GetImmatureCredit(bool fUseCache) const
+CAmount CWalletTx::GetImmatureCredit(const std::optional<int>& asOfHeight, bool fUseCache) const
 {
-    if (IsCoinBase() && GetBlocksToMaturity() > 0 && IsInMainChain())
+    if (IsCoinBase() && GetBlocksToMaturity(asOfHeight) > 0 && IsInMainChain(asOfHeight))
     {
         if (fUseCache && fImmatureCreditCached)
```

Comment:
Does the cache take into account `asOfHeight`? It doesn't seem to.

Reply:
Oh I see, `CWallet::GetCredit` only depends on the sum of values of txouts in the transaction that are "mine", which isn't dependent on the as-of height.